client/Containers/Main.jsx

```diff
@@ -2,19 +2,42 @@ import React, { Component } from 'react';
 import UserDashboard from './UserDashboard.jsx';
 import { connect } from 'react-redux';
 import TestProductList from './TestForProductList.jsx';
+import {
+  HashRouter as Router,
+  Route,
+  Switch,
+  Link,
+  useParams,
+} from 'react-router-dom';
+import AllProducts from './AllProducts';
+import MainNav from './MainNav';
+import FeaturedButton from '../components/styles/FeaturedButton';
 
 class Main extends Component {
   render() {
     const { fetchProductList } = this.props;
     return (
-      <div>
-        <hr />
-        <UserDashboard />
-        <hr className="heavy" />
-        <TestProductList />
-      </div>
+      <Router>
+        <div>
+          <Route component={MainNav} />
+          <Route component={AllProducts} path="/api/product/all" exact />
+          <Route component={UserDashboard} path="/api/login" exact />
+          <hr />
+          <Link to="/api/product/all">
+            <FeaturedButton> Shop All Furniture </FeaturedButton>
+          </Link>
+
+          <hr />
+          <hr className="heavy" />
+          <TestProductList />
+        </div>
+      </Router>
     );
   }
 }
 
-export default connect(null, null)(Main);
+const mapDispatchToProps = (dispatch) => ({
+  fetchProductList: () => dispatch(fetchProductList()),
+});
+
+export default connect(null, mapDispatchToProps)(Main);
```

Comment:
Can you undo this change? We should no longer need to connect Main to the `fetchProductList` thunk